### ISSUE TITLE
Request for fix false positive MissingFieldInEquals 

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#64]
false positive with a return name1.equals, test case:
```
class EqTryoutThisThat {
    final String name1;
    final String name2;

    public EqTryoutThisThat(String n1, String n2) {
        name1 = n1;
        this.name2 = n2;
    }
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        EqTryoutThisThat that = (EqTryoutThisThat) o;
        return this.name1.equals(that.name1) && name2.equals(that.name2);
    }
}
```

**Comment by jborgers** [comment]
fixed
